databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py

```diff
@@ -0,0 +1,135 @@
+        Dashboard name (Workbook name)
+        Dashboard description (Workbook description)
+        Dashboard creation timestamp (Workbook creationstamp)
+        Dashboard group name (Workbook top-level folder name)
+    As with all the Tableau extractors, uses the Metadata API: https://help.tableau.com/current/api/metadata_api/en-us/index.html
+    """
+
+    API_VERSION = const.API_VERSION
+    TABLEAU_HOST = const.TABLEAU_HOST
+    SITE_NAME = const.SITE_NAME
+    TABLEAU_ACCESS_TOKEN_NAME = const.TABLEAU_ACCESS_TOKEN_NAME
+    TABLEAU_ACCESS_TOKEN_SECRET = const.TABLEAU_ACCESS_TOKEN_SECRET
+    EXCLUDED_PROJECTS = const.EXCLUDED_PROJECTS
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+
+        self._conf = conf
+        self._auth = TableauDashboardAuth(self._conf)
+        self.query = """query {
+            workbooks {
+                id
+                name
+                createdAt
+                description
+                projectName
+                projectVizportalUrlId
+                vizportalUrlId
+            }
+        }"""
+
+        self._extractor = self._build_extractor()
+
+        transformers = []
+        timestamp_str_to_epoch_transformer = TimestampStringToEpoch()
+        timestamp_str_to_epoch_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, timestamp_str_to_epoch_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict({FIELD_NAME: 'created_timestamp', })))
+        transformers.append(timestamp_str_to_epoch_transformer)
+
+        dict_to_model_transformer = DictToModel()
+        dict_to_model_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, dict_to_model_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict(
+                    {MODEL_CLASS: 'databuilder.models.dashboard.dashboard_metadata.DashboardMetadata'})))
+        transformers.append(dict_to_model_transformer)
+        self._transformer = ChainedTransformer(transformers=transformers)
+
+    def extract(self):
+        # type: () -> Any
+
+        record = self._extractor.extract()
+        if not record:
+            return None
+
+        return self._transformer.transform(record=record)
+
+    def get_scope(self):
+        # type: () -> str
+
+        return 'extractor.tableau_dashboard_metadata'
+
+    def _build_extractor(self):
+        # type: ( -> TableauGraphQLApiMetadataExtractor
+        """
+        Builds a TableauGraphQLApiMetadataExtractor. All data required can be retrieved with a single GraphQL call.
+        :return: A TableauGraphQLApiMetadataExtractor that provides core dashboard metadata.
+        """
+        extractor = TableauGraphQLApiMetadataExtractor()
+        tableau_extractor_conf = \
+            Scoped.get_scoped_conf(self._conf, extractor.get_scope())\
+                  .with_fallback(self._conf)\
+                  .with_fallback(ConfigFactory.from_dict({TableauDashboardAuth.AUTH: self._auth,
```

Comment:
I didn't see this before, but it makes sense: Can we move all auth settings into `TableauGraphQLApiExtractor`? We don't use `self._auth` anywhere else in these top-level extractors, and it would cut out another config that we're passing around.

Reply:
Do you mean all the functionality, or just the settings constants? Given that we might want to make use of the REST API in the future, I think it would make more sense to keep them separate either way, if I'm understanding correctly

Reply:
The functionality. As it stands now, the auth token is set as `self._auth` but we don't need to authenticate until the GraphQL extractor, so instead of passing it as a config here, just do it in TableauGraphQLApiExtractor. That sets us up better to call that base extractor directly in the future.

Reply:
Oh, you mean deferring the instantiation of the `TableauDashboardAuth` object until the `init` of the `TableauGraphQLApiExtractor`. I thought you originally meant directly coupling the `auth` logic to the `TableauGraphQLApiExtractor` instead of just waiting to instantiate. Yeah that makes more sense to me, I'll change that